lib/rubocop/cop/minitest/assert_raises_with_regexp_argument.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # Checks for `assert_raises` with argument of regular expression literal.
+      # Arguments should be exceptions.
+      # The raised exception is returned and can be used
+      # to match against a regular expression.
+      #
+      # @example
+      #
+      #   # bad
+      #   assert_raises FooError, /some message/ do
+      #     obj.occur_error
+      #   end
+      #
+      #   # good
+      #   exception = assert_raises FooError do
+      #     obj.occur_error
+      #   end
+      #   assert_match(/some message/, exception.message)
+      #
+      class AssertRaisesWithRegexpArgument < Base
+        MSG = 'Do not pass regular expression literals to assert_raises. Test the resulting exception.'
```

Comment:
Can you tweak the message and insert `RESTRICT_ON_SEND` constant?

```suggestion
        MSG = 'Do not pass regular expression literals to `assert_raises`. Test the resulting exception.'
        RESTRICT_ON_SEND = %i[assert_raises].freeze
```